### PR TITLE
Tbrands 49 single column block refactor

### PR DESCRIPTION
**Pull request by JackHowa**
## Description

Set the stack with the role main as global 16 (desktop) and 11 (default)

- [x] Update block 
- [x] Create token file update in the block storybook
- [x] Create token file update per our process in the source of truth feature pack https://github.com/WPMedia/arc-themes-feature-pack/pull/232

## Jira Ticket

- [TBRANDS-49](https://arcpublishing.atlassian.net/browse/TBRANDS-49)

## Acceptance Criteria

Based on the figma designs apply the global token - spacing-16 as the gap of the single-column-regular to ensure all blocks placed on the page or spaced as per the designs

Set the gap within the Theme file to allow customisation

### Storybook 

1. Checkout this branch `git checkout TBRANDS-49-single-column-block-refactor`
2. `npm run storybook`
3. Go to the single column layout stories http://localhost:60001/?path=/story/layouts-single-column-regular-layout--layout-with-two-children
4. Notice the gap between the two children match the mock screenshotted in the ticket in desktop breakpoint

Screen Shot 2022-05-10 at 10 10 30

5. Decrease screen size to mobile. Then see the gap size change to 72 px per the mocks 

Screen Shot 2022-05-09 at 09 53 45

### Live fusion 

1. `git checkout TBRANDS-49-single-column-layout` feature pack 
2. `npx fusion start -f -l @wpmedia/single-column-layout-block`
3. Create a page with a single column layout
4. Put two blocks within the main to see the gap on mobile and desktop breakpoints
Screen Shot 2022-05-09 at 10 12 42
Screen Shot 2022-05-09 at 10 13 52

### Before

Feature blocks within dictate their spacing between each other

### After

Layout blocks dictate the blocks within themselves

## Dependencies or Side Effects

- none

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

**Comment by JackHowa** [comment]
Question: 

- In the ticket, I noticed a gap specified for only the regular column's stack. Does this ticket require setting a stack and its gap for the footer and the header as well?

File: blocks/single-column-layout-block/_index.scss [hunk 1]
```diff
@@ -9,6 +9,7 @@
 	width: 100%;
 	margin: auto;
 	@include scss.block-properties("single-column-regular");
+	@include scss.block-components("single-column-regular");
 }
 
 .b-single-column-regular-footer {
```
Comment (nschubach) [inline blocks/single-column-layout-block/_index.scss]: Should we BEM these classes?

**Comment by JackHowa** [comment]
> Question:
> 
> * In the ticket, I noticed a gap specified for only the regular column's stack. Does this ticket require setting a stack and its gap for the footer and the header as well?

Update after talking with Nick: 

- We want to make the header and footer to be stacks as well as the parent fragment. https://github.com/WPMedia/arc-themes-blocks/pull/1334/commits/3a987d87fe3739d09affb9b5b3f9c562b5eb50ee
- I will work on naming these using BEM standards, including the parent `div` 
- I set the roles based on their previous element used footer and header

File: blocks/single-column-layout-block/_index.scss (hunk 1 above)
Comment (JackHowa) [reply blocks/single-column-layout-block/_index.scss]: will work on this, good point 👀

Comment (JackHowa) [reply blocks/single-column-layout-block/_index.scss]: cool updated -- thanks https://github.com/WPMedia/arc-themes-blocks/pull/1334/commits/68e8ea61f0bbfe63e3cd0edfa06f2265bf0a51b4

File: blocks/single-column-layout-block/_index.scss [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
-	@include scss.block-properties("single-column-regular");
-}
+	&__header {
+		@include scss.block-properties("single-column-regular-header");
+		@include scss.block-components("single-column-regular-header");
+	}
+
+	&__main {
+		max-width: 90rem; // 1440px
+		width: 100%;
```
Comment (nschubach) [inline blocks/single-column-layout-block/_index.scss:11]: Should we move this into the theme CSS and use the new alias values or does it make sense to keep this here?

Comment (JackHowa) [reply blocks/single-column-layout-block/_index.scss:11]: I wasn't sure what was reasonable here for keeping. I can move this to the theme file if that's the recommendation

Comment (JackHowa) [reply blocks/single-column-layout-block/_index.scss:11]: the alias token layout-max-width?

Comment (JackHowa) [reply blocks/single-column-layout-block/_index.scss:11]: updated to move the scss to the token file

File: blocks/single-column-layout-block/layouts/single-column-regular/default.jsx [hunk 3]
```diff
@@ -1,21 +1,33 @@
...
+	if (!navigation && !main && !footer) {
+		return null;
+	}
 	return (
-		<>
-			{navigation ? <header className={`${LAYOUT_CLASS_NAME}-header`}>{navigation}</header> : null}
+		<Stack className={LAYOUT_CLASS_NAME}>
+			{navigation ? (
+				<Stack role="banner" className={`${LAYOUT_CLASS_NAME}__header`}>
+					{navigation}
```
Comment (nschubach) [inline blocks/single-column-layout-block/layouts/single-column-regular/default.jsx]: We just talked about this in the other ticket, but we should probably make these match as well.

File: blocks/single-column-layout-block/_index.scss (hunk 2 above)
Comment (matthewroach) [reply blocks/single-column-layout-block/_index.scss:11]: I don't think we should move this to the token file - we can make an assumption that the width is always 100% - a client could set a max-width to control the maximum width

File: blocks/single-column-layout-block/layouts/single-column-regular/default.jsx (hunk 3 above)
Comment (JackHowa) [reply blocks/single-column-layout-block/layouts/single-column-regular/default.jsx]: updated https://github.com/WPMedia/arc-themes-blocks/pull/1334/commits/2fbb40d57f73f891ae4946b61e189a87ac530b14

File: blocks/single-column-layout-block/_index.scss (hunk 2 above)
Comment (JackHowa) [reply blocks/single-column-layout-block/_index.scss:11]: will revert my move to the theme token file

Comment (JackHowa) [reply blocks/single-column-layout-block/_index.scss:11]: updated [3f25019](https://github.com/WPMedia/arc-themes-blocks/pull/1334/commits/3f250196c46bc05201a2c880fdf0b6643cb40770) [30ffa8d](https://github.com/WPMedia/arc-themes-blocks/pull/1334/commits/30ffa8d4205f83df37601cac7c260b1502f73afe)

File: blocks/single-column-layout-block/package.json [hunk 4]
```diff
@@ -20,6 +19,7 @@
     "directory": "blocks/single-column-layout-block"
   },
   "peerDependencies": {
-    "@arc-fusion/prop-types": "^0.1.5"
+    "@arc-fusion/prop-types": "^0.1.5",
+    "@wpmedia/arc-themes-components": "*"
   }
-}
+}
```
Comment (nschubach) [inline blocks/single-column-layout-block/package.json]:
```suggestion
}

```

File: blocks/single-column-layout-block/layouts/.npmignore [hunk 5]
```diff
@@ -1,3 +1,4 @@
 *.test.js
 *.test.jsx
-mock*.*
\ No newline at end of file
+mock*.*
+*.story.jsx
```
Comment (nschubach) [inline blocks/single-column-layout-block/layouts/.npmignore]:
```suggestion
*.story.jsx

```

**Comment by JackHowa** [comment]
@nschubach updated extra spacing

**Review by nschubach — APPROVED** [review]
Everything looks good to me.

**Comment by JackHowa** [comment]
looks slightly weird in storybook but the same behavior of max-width the 1400 seems to be there 

Screen Shot 2022-05-10 at 12 25 30
 

-> this is due to me removing the hard-coded width of 90 rem from the actual item itself https://github.com/WPMedia/arc-themes-blocks/pull/1334/files#diff-0ad9d2cc216848d4a848c230c716cf8115951c46e4422f13a9673622b703d6fcL28

**Pull request merged**